### PR TITLE
Guide on Forge contribution

### PR DESCRIPTION
**Pull request by PaleoCrafter**
As per @mezz's request, I wrote down a guide on how to set up a Forge development environment and I included his PR guidelines ([see forum](http://www.minecraftforge.net/forum/index.php?topic=43579.0)).

The guide currently only covers the setup for IntelliJ IDEA based on cpw's [video](https://www.youtube.com/watch?v=yanCpy8p2ZE). Hence, this PR is to be considered WIP.

ToDo list:
- [x] Add instructions for test mods to IDEA section
- [x] Add instructions for eclipse

### CODE REVIEW

File: docs/forgedev/index.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+If you're more of a visual person, our community's very own cpw has uploaded [a video](https://www.youtube.com/watch?v=yanCpy8p2ZE) explaining very similar steps which will also lead to a working setup.
+
+!!! Note
+    These steps will only work reliably from IDEA version 2016 onwards. Older versions didn't have the appropriate Gradle support and did not support Forge development workspaces.
+
+ 1. Import Forge's `build.gradle` as an IDEA project. For this, simply click `File -> Open`, then navigate to your fork's clone and select the `build.gradle` file.
+    If a dialog pops up, select "Open as Project".
+ 2. In the wizard that follows, make sure that "Create separate module per source set" is checked and that the "Use default gradle wrapper" option is active. Confirm the dialog.
+ 3. After IDEA is done importing the project and indexing the files, open the Gradle sidebar on the right hand side of your screen
+ 4. Open the "forge" project tree, select "Tasks", then "forgegradle" and right click the "Create Forge [setup]" option
```
Comment (liach) [inline docs/forgedev/index.md]: I would suggest to run `./gradlew setupForge` in command line. It is way slower to execute in idea.

**Comment by liach** [comment]
Status?

**Comment by PaleoCrafter** [comment]
I'll have some more free time in the next couple of days and I'll dedicate it to the various projects I've taken up recently, including this. So expect some progress in the near future.

**Comment by tterrag1098** [comment]
Any progress on this?

**Comment by PaleoCrafter** [comment]
So... I've finally found time to revisit this and added the instructions for getting test mods to work with IDEA. I still have to look into doing this with eclipse, but I'll ask people with some more experience with the IDE for that.

**Comment by mezz** [comment]
Thanks @PaleoCrafter !

**Pull request merged**